src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+        //                              Vdot_src = Q_src / (rho_src * Cp_src * DeltaT_src)
+        tmpSourceVolFlow = designSourceSideHeatTransfer / (rhoSrc * CpSrc * DeltaT_src);
+    } else if (!this->sourceSideDesignVolFlowRateWasAutoSized && this->sourceSideDesignVolFlowRate > 0.0) {
+        // given the value by the user
+        // set it directly
+        tmpSourceVolFlow = this->sourceSideDesignVolFlowRate;
+    } else if (!this->sourceSideDesignVolFlowRateWasAutoSized && this->sourceSideDesignVolFlowRate == 0.0) { // LCOV_EXCL_LINE
+        // user gave a flow rate of 0
+        // protected by the input processor to be >0.0
+        // fatal out just in case
+        errorsFound = true; // LCOV_EXCL_LINE
+        ShowSevereError(state,
+                        format("Invalid condenser flow rate for EIR PLHP (name={}; entered value: {}",
+                               this->name,
+                               this->sourceSideDesignVolFlowRate)); // LCOV_EXCL_LINE
+    } else {
+        // can't imagine how it would ever get to this point
+        // just assume it's the same as the load side if we don't have any sizing information
+        tmpSourceVolFlow = tmpLoadVolFlow; // LCOV_EXCL_LINE
+    }
+
+    this->sourceSideDesignVolFlowRate = tmpSourceVolFlow;
+
+    if (errorsFound) {
+        ShowFatalError(state, "Preceding sizing errors cause program termination"); // LCOV_EXCL_LINE
+    }
+}
+
+void EIRFuelFiredHeatPump::resetReportingVariables()
+{
+    this->loadSideHeatTransfer = 0.0;
+    this->loadSideEnergy = 0.0;
+    this->loadSideOutletTemp = this->loadSideInletTemp;
+    this->powerUsage = 0.0;
+    this->powerEnergy = 0.0;
+    this->fuelUsage = 0.0;
+    this->fuelEnergy = 0.0;
+    this->sourceSideHeatTransfer = 0.0;
+    this->sourceSideOutletTemp = this->sourceSideInletTemp;
+    this->sourceSideEnergy = 0.0;
+}
+
+PlantComponent *EIRFuelFiredHeatPump::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType hp_type_of_num, const std::string &hp_name)
+{
+    if (state.dataEIRFuelFiredHeatPump->getInputsFFHP) {
+        EIRFuelFiredHeatPump::processInputForEIRPLHP(state);
+        EIRFuelFiredHeatPump::pairUpCompanionCoils(state);
+        state.dataEIRFuelFiredHeatPump->getInputsFFHP = false;
+    }
+
+    for (auto &plhp : state.dataEIRFuelFiredHeatPump->heatPumps) {
+        if (plhp.name == UtilityRoutines::MakeUPPERCase(hp_name) && plhp.EIRHPType == hp_type_of_num) {
+            return &plhp;
+        }
+    }
+
+    ShowFatalError(state, "EIR Fuel-Fired Heat Pump factory: Error getting inputs for PLFFHP named: " + hp_name);
+    return nullptr; // LCOV_EXCL_LINE
+}
+
+void EIRFuelFiredHeatPump::pairUpCompanionCoils(EnergyPlusData &state)
+{
+    for (auto &thisHP : state.dataEIRFuelFiredHeatPump->heatPumps) {
+        if (!thisHP.companionCoilName.empty()) {
+            auto thisCoilName = UtilityRoutines::MakeUPPERCase(thisHP.name);
```

Comment:
Some bad use of `auto` here. In this specific case you will get a `std::string` which is what you want, but in other places you may want a `std::string &` or a `std::string_view`. For strings, better to explicitly say what you want, even if it is `std::string`.